core/common/src/main/java/alluxio/heartbeat/HeartbeatScheduler.java

```diff
@@ -118,17 +118,13 @@ public static void schedule(String threadName) {
   }
 
   /**
-   * Waits until the given thread can be executed.
+   * Waits for up to 10 seconds until the given thread can be executed.
    *
    * @param name a name of the thread to wait for
    * @throws InterruptedException if the waiting thread is interrupted
    */
   public static void await(String name) throws InterruptedException {
-    try (LockResource r = new LockResource(sLock)) {
-      while (!sTimers.containsKey(name)) {
-        sCondition.await();
-      }
-    }
+    await(name, 10, TimeUnit.SECONDS);
```

Comment:
It is pretty weird to have anti-symmetry with `Condition.await()`. What do you think of changing the name to `waitForSchedulable`? If you want to follow a general rule of always having a method that does not timeout, I have some questions:

- Do you think we should include the version which does not timeout even if nothing uses it? If so, what's the advantage of having such a method? 
- When would it be useful for a test to hang forever instead of timing out after some period of time?
- How would you name a version of the method which waits for a default amount of time before timing out?